### ISSUE TITLE
Ecosia not being added into default SE list for "Hungary" or "Luxemburg" after upgrading

### ISSUE THREAD

**Issue by kjozwiak** [brave/brave-browser#13813]
## Description 

Ecoisa isn't being added into the SE default list for both `Hungary` or `Luxemburg` when upgrading from `1.19.86 CR: 88.0.4324.96` to `1.19.88 CR: 88.0.4324.96`. However, it's being added when installing `1.19.88 CR: 88.0.4324.96` (new profile).

@btlechowski reported a similar issue on Linux, however, Ecosia isn't being added on new installs either as https://github.com/brave/brave-browser/issues/13571 which is a different issue affecting Linux.

## Steps to Reproduce

   1. change region to either `Hungary` or `Luxemburg`
   2. install  `1.19.86 CR: 88.0.4324.96` and ensure Ecosia isn't in the SE list under `brave://settings/search`
   3. upgrade to `1.19.88 CR: 88.0.4324.96` and notice Ecosia missing from  `brave://settings/search`

## Actual result:

Ecosia isn't being added into the default SE when upgrading from `1.19.86 CR: 88.0.4324.96` to `1.19.88 CR: 88.0.4324.96`

## Expected result:

Ecosia should be added into the default SE list when upgrading from `1.19.86 CR: 88.0.4324.96` to `1.19.88 CR: 88.0.4324.96`

## Reproduces how often: 

100% reproducible using the STR mentioned above (tried several times)

## Brave version (brave://version info)

```
Brave | 1.19.88 Chromium: 88.0.4324.96 (Official Build) (64-bit)
-- | --
Revision | 68dba2d8a0b149a1d3afac56fa74648032bcf46b-refs/branch-heads/4324@{#1784}
OS | Windows 10 OS Version 2009 (Build 19042.746)
```

## Version/Channel Information:

- Can you reproduce this issue with the current release? `Yes`
- Can you reproduce this issue with the beta channel? `Yes`
- Can you reproduce this issue with the nightly channel? `Yes`

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? `N/A`
- Does the issue resolve itself when disabling Brave Rewards? `N/A`
- Is the issue reproducible on the latest version of Chrome? `N/A`

## Miscellaneous Information:

CCing @bsclifton @rebron @brave/legacy_qa

**Comment by kjozwiak** [comment]
@LaurenWags can you see if you can reproduce on `macOS`? @srirambv mind checking `Android`?

**Comment by srirambv** [comment]
Not reproducible on Android. Upgrading from 1.19.86 to 1.19.88 shows Ecosia in the SE list and can be set as default. Performing a search via URL with Ecosia adds the referral code as expected


https://user-images.githubusercontent.com/17010094/106132728-38a2a200-615c-11eb-88e0-c4eaf4ce9fbd.mp4

**Comment by LaurenWags** [comment]
Luxemburg and Hungary are getting added on upgrade from `1.19.86` to `1.19.88` on macOS. Here's Luxemburg screenshots as an example:

Region | `1.19.86` | `1.19.88`
----- | ----- | -----
<img width="690" alt="Region" src="https://user-images.githubusercontent.com/28145373/106154306-72b87780-614d-11eb-87d5-8d6dfdde6237.png"> | <img width="1037" alt="1 19 86" src="https://user-images.githubusercontent.com/28145373/106154314-74823b00-614d-11eb-81b4-2d1a1558850b.png"> | <img width="1037" alt="1 19 88" src="https://user-images.githubusercontent.com/28145373/106154324-764bfe80-614d-11eb-906c-389ad4085254.png">

**Comment by srirambv** [comment]
Works on Linux as well upgrading from 1.19.86 to 1.19.88
![image](https://user-images.githubusercontent.com/17010094/106159137-6566b180-617c-11eb-8da6-78fb56f1654d.png)

https://bravesoftware.slack.com/archives/C015G1THCAJ/p1611848047014400?thread_ts=1611765253.011700&cid=C015G1THCAJ

**Comment by kjozwiak** [comment]
Thanks for checking @srirambv & @LaurenWags 👍 Going to run through it again on `Win 10 x64`. Might be isolated to that platform only. @GeetaSarvadnya @btlechowski mind trying to see if you can reproduce on `Win`?

**Comment by kjozwiak** [comment]
Double checked now that we released `1.19.88 Chromium: 88.0.4324.96` and seems like it's working without any issues. I tried both `Hungary` & `Luxemburg` via the stable update channel and `Ecosia` was being added.

Going to close this off as it's working. Apologies for the false red flag @bsclifton. Thanks for double checking @srirambv & @LaurenWags!
